apps/designer/app/designer/features/style-panel/sections/spacing/input-popover.tsx

```diff
@@ -0,0 +1,118 @@
+import { useState } from "react";
+import {
+  keyframes,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverPortal,
+  styled,
+} from "@webstudio-is/design-system";
+import {
+  CssValueInput,
+  type IntermediateStyleValue,
+} from "../../shared/css-value-input";
+import type { StyleProperty, StyleValue } from "@webstudio-is/css-data";
+import type { StyleChangeEvent } from "./types";
+
+const slideUpAndFade = keyframes({
+  "0%": { opacity: 0, transform: "scale(0)" },
+  "100%": { opacity: 1, transform: "scale(1)" },
+});
+
+const Input = ({
+  value,
+  property,
+  onChange,
+  onClosePopover,
+}: {
+  property: StyleProperty;
+  value: StyleValue;
+  onChange: (event: StyleChangeEvent) => void;
+  onClosePopover: () => void;
+}) => {
+  const [intermediateValue, setIntermediateValue] = useState<
+    StyleValue | IntermediateStyleValue
+  >();
+
+  return (
+    <CssValueInput
+      property={property}
+      value={value}
+      intermediateValue={intermediateValue}
+      onChange={(styleValue) => {
+        setIntermediateValue(styleValue);
+        if (styleValue.type !== "intermediate") {
+          onChange({ property, value: styleValue, isEphemeral: true });
+        }
+      }}
+      onHighlight={(styleValue) => {
+        onChange({ property, value: styleValue, isEphemeral: true });
+      }}
+      onChangeComplete={(styleValue, reason) => {
+        setIntermediateValue(undefined);
+        onChange({ property, value: styleValue, isEphemeral: false });
+
+        // @todo: handle Esc
+        if (reason === "blur" || reason === "enter") {
+          onClosePopover();
+        }
+      }}
+    />
+  );
+};
+
+// trigger is used only for positioning
+const Trigger = styled("div", { position: "absolute", width: 0, height: 0 });
+
+const PopoverContentStyled = styled(PopoverContent, {
+  minWidth: 0,
+  minHeight: 0,
+  width: 88,
```

Comment:
how did we end up with this one? can it use token?

Reply:
Width of the popover from Figma.

<img width="585" alt="Screenshot 2022-12-21 at 13 39 01" src="https://user-images.githubusercontent.com/825702/208873298-5ae11ba2-7efe-42c3-8496-03f7bcf04198.png">

We don't have tokes for neither 88 nor 72 (input width).

Reply:
that means something is wrong with the design, @taylornowotny probably just didn't use the right spacing, I would use the closest thing we have in tokens to it, he will update when he can

Reply:
Switched to `$spacing$20` which is `80px`. I think it looks good too:

<img width="281" alt="Screenshot 2022-12-21 at 14 18 34" src="https://user-images.githubusercontent.com/825702/208881735-2693bc3f-a657-435a-b065-3542a45689e8.png">